### PR TITLE
[#mhv-51045] added support for isRefillable field

### PR DESCRIPTION
**Pull request by KolbyVA**
## Summary
Fixes issue where the refill button was appearing on medications that were not refillable.
Solution is to use the isRefillable field that we were not previously using.

## Related issue(s)

> - https://jira.devops.va.gov/browse/MHV-51045
> - MHV is showing 10 active, of which 6 can be requested for fill/refill
> - RxR app showing 10 active, of which 6 can be requested for fill/refill
> - VAHB app showing 10 active, of which 6 can be requested for fill/refill
> - chatbot prescriptions
> - MHV on [va.gov](http://va.gov/) is showing 10 active of which 9 can be requested for fill/refill
> MHV on [va.gov](http://va.gov/) is showing more prescriptions that can be filled/refilled than the other medication tools

## Testing done
unit tests updated and passing.

## Screenshots
No styling changes, just fixed which medications show the button.

## What areas of the site does it impact?
mhv/medications

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by acrollet — CHANGES_REQUESTED** [review]
@KolbyVA quick work, awesome! 🚀  This looks good, my only request is if we could have a test assertion that proves that we don't show the refill button when `isRefillable == false`?

**Review by acrollet — APPROVED** [review]
🚀

**Review by vmenshutin-bylight — APPROVED** [review]

File: src/applications/mhv/medications/components/shared/FillRefillButton.jsx [hunk 1]
```diff
@@ -12,15 +12,11 @@ const FillRefillButton = rx => {
...
-    dispStatus,
     success,
+    isRefillable,
   } = rx;
 
-  if (
-    (dispStatus === 'Active' && refillRemaining !== 0) ||
-    dispStatus === 'Active: Parked'
-  ) {
+  if (isRefillable) {
```
Comment (vmenshutin-bylight) [inline src/applications/mhv/medications/components/shared/FillRefillButton.jsx:19]: Not sure it is safe here. Please take a look at FillRefillButton change history (refer to Slack chat)

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**